libcnb/examples/example-02-ruby-sample/src/layers/ruby.rs

```diff
@@ -1,82 +1,65 @@
-use std::collections::HashMap;
 use std::fs;
 use std::io;
 use std::path::Path;
 
 use flate2::read::GzDecoder;
-use libcnb::data::layer_content_metadata::{LayerContentMetadata, LayerTypes};
-use libcnb::layer_lifecycle::LayerLifecycle;
-
-use std::env;
 use tar::Archive;
 use tempfile::NamedTempFile;
 
 use crate::RubyBuildpack;
 use libcnb::build::BuildContext;
+use libcnb::data::layer_content_metadata::LayerTypes;
 use libcnb::generic::GenericMetadata;
+use libcnb::layer::{Layer, LayerResult, LayerResultBuilder};
+use libcnb::layer_env::{LayerEnv, ModificationBehavior, TargetLifecycle};
+
+pub struct RubyLayer;
+
+impl Layer for RubyLayer {
+    type Buildpack = RubyBuildpack;
+    type Metadata = GenericMetadata;
 
-pub struct RubyLayerLifecycle;
+    fn types(&self) -> LayerTypes {
+        LayerTypes {
+            build: true,
+            launch: true,
+            cache: false,
+        }
+    }
 
-impl LayerLifecycle<RubyBuildpack, GenericMetadata, HashMap<String, String>>
-    for RubyLayerLifecycle
-{
     fn create(
         &self,
+        context: &BuildContext<Self::Buildpack>,
         layer_path: &Path,
-        build_context: &BuildContext<RubyBuildpack>,
-    ) -> anyhow::Result<LayerContentMetadata<GenericMetadata>> {
+    ) -> anyhow::Result<LayerResult<Self::Metadata>> {
         println!("---> Download and extracting Ruby");
 
         let ruby_tgz = NamedTempFile::new()?;
 
         download(
-            &build_context.buildpack_descriptor.metadata.ruby_url,
+            &context.buildpack_descriptor.metadata.ruby_url,
```

Comment:
Fun fact: the type itself is currently called `BuildpackToml`, which is something @hone dislikes. I'm torn, in general I would argue that using the official names from the spec are the ones we should use. But I also get you point about discoverability, buildpack authors know what a `buildpack.toml` is, since it's the most prominent file they interact with.

Given this is nothing that changed in this PR, I'd say we leave it as-is for now and file an issue where we can discuss this outside of this PR. We need to at least make the naming consistent between the type and the field name.

Reply:
I filed an issue for this as #168